### PR TITLE
a simulation, INSIDE, of a simulation! NTSL updates

### PR DESCRIPTION
**Pull request by H0rologium**
## About The Pull Request

This PR attempts to complete the following goals: 
1. Fix the `broadcast` system (script-side) to actually function as intended.
2. Allow NTSL code blocks using process_signal() to optionally return two signal objects.
3. Added an optional argument to the signal constructor.
(Also I forgot to add 'monkey' to the language list in #527 , added it here. ooh ooh ah ah.)

**For point 1:** The broadcast function just was not working. It now is, you must create the signal object as its own variable in-script first. The broadcast function already was 'implemented', the function was designed to send a message but with control over the properties of the message from NTSL.
Example:
```
...
newsignal = signal("My legs have atrophied", 1355, "Aemil Ferrilinea", "Bartender");
broadcast(newsignal);
...
```
Note that the signal object has not fundamentally changed. Inputting invalid information will cause the broadcast to fail. Also, the broadcast will only be emitted when the code is returned (if you return newsignal) meaning it still follows telecommunication protocols. This includes loggings. I am not 100% certain but I believe that instantiating in the method call is a limitation of DM and indirectly the code behind NTSL.

**For point 2:** This was a limitation on how NTSL itself was coded. In the 'newer' revisions, you need to use process_signal to run your code. You currently cannot work around this at all, the code _will not compile in-game_ if you do not code in this method. This means older structures involving multiple method calls do not work. Another restriction in place (that is not changed) is that this method only returns a signal object (changed in this PR). 

What this PR does is allow the user to return up to two signal objects contained in a _list_. 
Example:
`return list(signalA,signalB);`

You can still call a return through `return signal;`, or even have a 1-element list such as `return list(signalA);`. The interpreter has been recoded to properly iterate through this list. **It is very important that I add: the interpreters eval of the list will only allow a maximum number of iterations equal to 2 (even if the list had more elements when returned). This is a hard value, meaning it can only be changed through the source code. All NTSL code still follows the same input sanitization processes as before.**

**Point 3:** Signal object construction. This object, for some reason, was missing the 'language' property. It does get added later in runtime, but I decided to add it in at initialization because that sounds better in my mind.

## Why It's Good For The Game

TG bad
ok now that's out of the way, this update to comms would be good as I am trying to restore health to one of the segments of TG code I feel is very injured at the moment. This will also help NTSL be a bit more flexible in how you can approach it.

The telecomms role(s) feel a bit like they are somewhat useless, and what they primarily do ends up just being messing with other people. Me and the other 3 NTSL coders would like to see an expansion to this role. Not necessarily adding a lot of busywork for these workers but allowing their job to actually interact with the station more than just 'fuck up general comms', "reconnect broken servers" and "telling the captain that the isometric distortion is the reason that the servers are down". This is a good step in that direction.

## Changelog
:cl:
add: Added a new method for code blocks to return two values.
add: Added more comments <3
tweak: tweaked how broadcast works.
removed: removed a commented part of the code floating around. It had been this way since TG afaik, definitely still works without it.
add: Added a new language argument to the constructor of the 'signal' object. 
server: alters NTSL code that is not part of the maps.
/:cl:

### CODE REVIEW

**Comment by Kitsunemitsu** [comment]
I don't know who you are, but thank you for assisting us in our NTSL endeavours.
When I ported NTSL from Yog/Rat I was going to have it as a small side project.
All the admins fucking despised me for it. I really doubted whether it would be any fun at all.
So I ported it, and it sat there as a side feature for a little while. I wasn't actually too sure how to code it, sadly.

But I'm really glad that you do.
An Unsung hero.

Please contact me on our discord for a contributor role.

In short, a PR that is well needed for one of our signature roles, by someone who knows the code far better than I do with no merge issues. 
Merging.
